components/raftstore/src/store/peer_storage.rs

```diff
@@ -357,6 +357,25 @@ pub fn recover_from_applying_state(
     Ok(())
 }
 
+pub fn init_applied_index_term(
+    engines: &Engines,
+    region: &Region,
+    apply_index: u64,
+) -> Result<u64> {
+    if apply_index == RAFT_INIT_LOG_INDEX {
+        return Ok(RAFT_INIT_LOG_TERM);
+    }
+    let state_key = keys::raft_log_key(region.get_id(), apply_index);
+    match engines.raft.get_msg::<Entry>(&state_key)? {
```

Comment:
It looks to me like after this patch, this get_msg might be called more frequently than in the other patch, and in cases where it is never used.

That is, previously, this statement was behind the else branch of `if idx == apply_state.get_truncated_state().get_index()` condition in https://github.com/tikv/tikv/pull/7119/commits/01e2e9b03479ca9310e659e69bbeb67fee9f6eca#diff-1cfebd95dd15d93f24508344639ce78aL1364.

Now it is executed unconditionally.

Is that a potential performance problem?

Reply:
No. I just do it for remove raftdb from apply.rs. It will make my job for abstract more easy.